Create a positive test for `is_key_header`

mercybassey opened the pull request:
Thank you for taking your time to contribute to Ersilia, just a few checks before we proceed
- [x] Have you followed the guidelines in our [Contribution Guide](https://github.com/ersilia-os/ersilia/blob/master/CONTRIBUTING.md)
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

**Description**

Test that `is_key_header` identifies case variations of `key_header_synonyms`.

**Changes to be made**

Create a positive test case for `is_key_header`.

**Status**

This PR creates a positive test case for `is_key_header`.

Is this pull request related to any open issue? If yes, replace issueID below with the issue ID

Related to #1319

DhanshreeA commented on test/compound_test_identifier.py:
Hi @mercybassey thank you for your work! Could you put this test in the `test_compound_identifier.py` file instead? See it [here](https://github.com/ersilia-os/ersilia/blob/be2a3f654a710e96bca06c9b1c1ad2dbd01d4bce/test/test_compound_identifier.py).

DhanshreeA commented on test/compound_test_identifier.py:
```diff
@@ -0,0 +1,15 @@
+def compound_identifier():
+    return CompoundIdentifier()
+
+def test_is_key_header_positive(compound_identifier):
```
I would also recommend looking into [parameterization](https://docs.pytest.org/en/stable/how-to/parametrize.html) in pytest fixtures. @musasizivictoria has done a wonderful job to start us off, you could work together!

mercybassey replied:
Sure.

mercybassey commented:
@DhanshreeA I have incorporated parameterization in the test case. Initially, I wrote the test to check for uppercase, lowercase, and mixed-case inputs. However, I have now removed mixed-case inputs for the sake of consistency; so it matches with @musasizivictoria test case.
